pkg/kubeadm/master/addons.go

```diff
@@ -101,6 +105,9 @@ func createKubeDNSPodSpec(s *kubeadmapi.KubeadmConfig) api.PodSpec {
 	)
 
 	return api.PodSpec{
+		NodeSelector: map[string]string{
+			"beta.kubernetes.io/arch": runtime.GOARCH,
```

Comment:
Above you're using `architecture` variable but here you're using something different. Why?